src/merkle/index.rs

```diff
@@ -140,4 +166,51 @@ mod tests {
             }
         }
     }
+
+    /// Test that using `leaf_to_root` or `move_up` give the same result.
+    #[test]
+    fn test_node_index_and_leaf_to_root_match() {
+        let value = 0b10101010_01010101_11110000_10110111;
+
+        for depth in 1..=64 {
+            let mut index = NodeIndex::new(depth, value);
+
+            let bits: Vec<Direction> = index.leaf_to_root().collect();
+            let mut indexes: Vec<Direction> = Vec::new();
+            for _ in 0..depth {
+                indexes.push(index.direction());
+                index.move_up();
+            }
+
+            assert_eq!(bits, indexes);
+        }
+    }
+
+    /// Test with the same value and different depths, the smaller depth is a prefix of the larger
+    /// one (this allows nodes to be moved up and down in a SMT).
+    #[test]
+    fn test_increasing_depth_doesnt_change_prefix() {
+        let value = 0b10101010_11110000_10110111_01010101;
+
+        // accumulates the new bit on every iteration
+        let mut prefix: Vec<Direction> = Vec::new();
+        for depth in 1..=64 {
+            let mut index = NodeIndex::new(depth.into(), value);
+
+            let bits: Vec<Direction> = index.leaf_to_root().collect();
```

Comment:
changing this to `root_to_leaf` will break the test, which is the stability issue